posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -171,67 +204,36 @@ def get_results_iterator(
     if str(team_id) in settings.UNCONSTRAINED_TIMESTAMP_TEAM_IDS:
         timestamp_predicates = ""
 
+    if fields is None:
+        query_fields = ",".join((f"{field['expression']} AS {field['alias']}" for field in default_fields()))
+    else:
+        query_fields = ",".join((f"{field['expression']} AS {field['alias']}" for field in fields))
+
     query = SELECT_QUERY_TEMPLATE.substitute(
-        fields=S3_FIELDS if include_person_properties else FIELDS,
-        order_by="ORDER BY inserted_at",
+        fields=query_fields,
+        order_by="ORDER BY COALESCE(inserted_at, _timestamp)",
```

Comment:
note: This should actually fix a bug as values without `inserted_at` were all sorted together.